### ISSUE TITLE
Cannot read properties of undefined (reading 'odometer')

### ISSUE THREAD

**Issue by UntappedSolutions** [inferusvv/react-odometerjs#46]
I am using react and gatsby. I have followed the gatsby instructions and all seems to be working, but when i include the  <Odometer value={1234} format="(ddd),dd" /> line within the rendered html I get the following error. Any ideas? @inferusvv  inferusvv

<img width="988" alt="Screenshot 2023-04-28 at 13 04 29" src="https://user-images.githubusercontent.com/31103744/235142956-7c57d7cc-c8ce-49ce-be05-54e7df7b6a7f.png">

**Comment by UntappedSolutions** [comment]
Hi @inferusvv, I just wanted to see if there was any feedback on this issue? Is this something you have come across before? Any help much appreciated. Thank you!

**Comment by bezenson** [comment]
> Hi @inferusvv, I just wanted to see if there was any feedback on this issue? Is this something you have come across before? Any help much appreciated. Thank you! 

I will try to check soon. Could you create a demo on code sandbox for example?

**Comment by UntappedSolutions** [comment]
Thank you for getting back to me. Its a very simple example. Shall I record a short video of recreating the issue, im sure its something really obvious I am missing. Or maybe a 10 min pair session, whichever you prefer

**Comment by UntappedSolutions** [comment]
@inferusvv Incase it is something simple, I have a gatsby & react app set up. I used yarn to install react-odometerjs, when importing the node_module directly I get the "Failed to Server Render (SSR). Error message: ReferenceError: document is not defined" error which is why the gatsby section in your docs exists I guess.  
I have then added loadable, and using the Odometer.js file you mention in the gatsby docs. 

It loads fine but then when the script runs client-side to pick up the elements and animate them I get the error above. 

I have the example:
```
      <div className="odometer-container">
        <Odometer value={1234} format="(ddd),dd" />
      </div>
```

given my example is such a simple use of your script I thought maybe you had come across options.el being undefined when you wrote the gatsby docs?

**Comment by UntappedSolutions** [comment]
I also found the below 2 errors at the top of the console log which I believe to be the route cause. I am not sure what the fix would be and why this would be any different to the other gatsby implementations of react- odometerjs

<img width="1057" alt="Screenshot 2023-05-09 at 23 50 51" src="https://github.com/inferusvv/react-odometerjs/assets/31103744/d9ce6097-c157-4705-8d7e-fcac63b50964">
<img width="1057" alt="Screenshot 2023-05-09 at 23 51 14" src="https://github.com/inferusvv/react-odometerjs/assets/31103744/048671a9-eced-4f3c-95ec-007ff9caa192">

**Comment by bezenson** [comment]
@UntappedSolutions  I think, that I need a demo from your side. Maybe create a public repository.

Here is my code:

```js
import * as React from "react"
import type { HeadFC, PageProps } from "gatsby"
import loadable from "@loadable/component"
import "../styles/odometer.css"

const Odometer = loadable(() => import('react-odometerjs'));

const pageStyles = {
  color: "#232129",
  fontSize: "24px",
  padding: 96,
  fontFamily: "-apple-system, Roboto, sans-serif, serif",
}

const IndexPage: React.FC<PageProps> = () => {
  const [count, setCount] = React.useState(1000);
  React.useEffect(() => {
    const timeoutId = setTimeout(() => {
      setCount(4321);
    }, 1000);
    return () => clearTimeout(timeoutId);
  }, []);
  return (
    <main style={pageStyles}>
      <Odometer duration={1000} value={count} />
    </main>
  )
}

export default IndexPage

export const Head: HeadFC = () => <title>Home Page</title>
```

And here is a result:

https://github.com/inferusvv/react-odometerjs/assets/3227120/d7185f29-7764-4cc2-89e1-29f4ac50d17f

**Comment by UntappedSolutions** [comment]
Hi @inferusvv, I notice that you are using `const Odometer = loadable(() => import('react-odometerjs'));` where as I followed the Gatsby docs which suggested creating an extra file with the following inside: 
`import Odometer from 'react-odometerjs'

export default Odometer;`

This has solved the immediate error.

I have now got it working. I needed to download the css and import it directly, not a big issue. 

I'd suggest the key update to the docs would be to use loadable to load the node module and not the extra file. Thank you for your messages, they really helped get to the bottom of it quickly.
